--- zincati-0.0.3/Cargo.toml	1970-01-01T00:00:00+00:00
+++ zincati-0.0.3/Cargo.toml	2019-07-10T19:11:21.227014+00:00
@@ -39,7 +39,7 @@
 version = "^0.4.7"
 
 [dependencies.env_logger]
-version = "^0.6.2"
+version = "^0.6.1"
 
 [dependencies.envsubst]
 version = "^0.1.0"
@@ -48,7 +48,7 @@
 version = "^0.1.5"
 
 [dependencies.futures]
-version = "^0.1.28"
+version = "^0.1.27"
 
 [dependencies.glob]
 version = "^0.3.0"
